autoswitch_virtualenv.plugin.zsh

```diff
@@ -181,11 +182,19 @@ function check_venv()
             file_permissions="$(/usr/bin/stat -f %OLp "$venv_path")"
         fi
 
-        if [[ "$file_owner" != "$(id -u)" ]]; then
+        if [[ -z $AUTOSWITCH_DISABLE_WARNING ]] || [ "$AUTOSWITCH_DISABLE_WARNING" = 0 ]; then
+            # Security not disabled
+            security_is_on=1
+        else
+           # Security disabled
+            security_is_on=0
+        fi
```

Comment:
I would abstract this into a function. Something along the lines of 

```shell
function _autoswitch_warnings_are_disabled() {
    if [[ -z $AUTOSWITCH_DISABLE_WARNING ]] || [ "$AUTOSWITCH_DISABLE_WARNING" = 0 ]; then
        # Security not disabled
        printf "true"
    else
        printf "false"
    fi
}
```

and then in the code

```shell
elif ! { [[ "$file_permissions" =~ ^[64][04][04]$ ]] || [ "$(_autoswitch_warnings_are_disabled" = "false" ]; }
```

This where shell scripting feels very unnatural to use because I'm not sure if I prefer a stringified true/false or a stringified 0/1